.github/workflows/codecov.yaml

```diff
@@ -14,13 +14,26 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Setup Sui CLI
-        uses: ./.github/actions/install-sui-cli
+      # This step ensures that the "LITERAL" value is given to the actions.
+      # Using `jq` sends it uninterepted to the action which is sort of "hackish" and doesn't ensure the separation of concerns.
+      - name: Get SUI and NodeJS Versions
+        shell: bash
+        run: |
+          SUI_VERSION=$(jq -r '.SUI_VERSION' "version.json")
+          echo "SUI_VERSION=$SUI_VERSION" >> $GITHUB_ENV
+
+          NODEJS_VERSION=$(jq -r '.engines.node' "package.json")
+          echo "NODEJS_VERSION=$NODEJS_VERSION" >> $GITHUB_ENV
```

Comment:
this duplication everywhere doesn't make sense. make it read this in the node action. The node action can underthehood call the more generic action from another repo that you were mentioning in your other PR